### ISSUE TITLE
warning: statement is unreachable

### ISSUE THREAD

**Issue by neoblizz** [Tencent/rapidjson#1545]
```
rapidjson/include/rapidjson/writer.h(552): warning: statement is unreachable
rapidjson/include/rapidjson/writer.h(353): warning: statement is unreachable
```

How do we resolve this?

**Comment by fredgan** [comment]
It seems that it can not match the source code. Did you modify the file?

**Comment by neoblizz** [comment]
@fredgan no, I did not. It may be due to the gcc/g++ version difference, could you please tell me which version are you using?

**Comment by fredgan** [comment]
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.11)

**Comment by neoblizz** [comment]
I have a few versions that I have tried it on, this being the current one:
gcc (Ubuntu 5.5.0-12ubuntu1~16.04) 5.5.0 20171010

And in my setting, it is being included in a project with c++11 flag.
